pkg/controller/datadogagentdeployment/agent.go

```diff
@@ -111,7 +112,7 @@ func (r *ReconcileDatadogAgentDeployment) deleteDaemonSet(logger logr.Logger, ag
 		return err
 	}
 	logger.Info("Delete DaemonSet", "daemonSet.Namespace", ds.Namespace, "daemonSet.Name", ds.Name)
-	r.recorder.Event(agentdeployment, corev1.EventTypeNormal, "Delete DaemonSet", fmt.Sprintf("%s/%s", ds.Namespace, ds.Name))
+	r.recordEvent(agentdeployment, corev1.EventTypeNormal, "Delete DaemonSet", fmt.Sprintf("%s/%s", ds.Namespace, ds.Name), datadog.DeletionEvent)
```

Comment:
maybe we can move the `fmt.Sprintf()` inside the function and just request an `ObjectMeta` parameter instead

Reply:
I agree that we need to reduce the number of arguments, but internal objects don't implement the `NameNamespace` interface (missing `GetKind()` method). We might need to use an intermediate struct to pass event data to `recordEvent`

Reply:
also we always pass `corev1.EventTypeNormal` as `eventtype`, maybe we can remove the argument use the value directly in `recordEvent`